test/integration/utils.test.js

```diff
@@ -189,4 +189,20 @@ describe(Support.getTestDialectTeaser('Utils'), function() {
       expect(Utils.singularize('status')).to.equal('status');
     });
   });
+
+  describe('formatReferences', function () {
+    ([
+      [{referencesKey: 1}, {references: {model: undefined, key: 1}}],
+      [{references: 'a'}, {references: {model: 'a', key: undefined}}],
+      [{references: 'a', referencesKey: 1}, {references: {model: 'a', key: 1}}],
+      [{references: {model: 1}}, {references: {model: 1}}]
+    ]).forEach(function (test) {
+      var input  = test[0];
+      var output = test[1];
+
+      it('converts ' + JSON.stringify(input) + ' to ' + JSON.stringify(output), function () {
+        expect(Utils.formatReferences(input)).to.deep.equal(output);
+      });
+    });
+  });
```

Comment:
> The test itself though is pretty useless right now

Strong words. Calm down. The tests are testing the helper. Period. They do exactly what they supposed to do. I will move it to the unit tests though.